free-2.md

```diff
@@ -11,41 +11,81 @@ _What need is met by your idea? Who are you helping? What is that person's
+
+
 ### Operation (semantics)
 
 _What might happen when a program runs? How does a program interact with the
 user? What kinds of errors might occur, and how might they be communicated to
 the user?_
 
+Since this DSL involves two distinct data types, I will assume that it will be built on a statically typed language to help users catch mistakes early. Programming in this DSL will likely occur within the host language’s IDE (instead of a terminal or a separate GUI) to leverage existing tools for code formatting and type checking. 
+
+The high-level order of actions when a program run is as follows:
+- Start at the code block associated with composition (similar to Java’s main function)
+- Check that the given files exist and are of the correct file types
+- Some kind of conversion turns those files into data types that the DSL can more easily operate on
+- Apply the programmed transformations onto the DSL’s representation of a blank canvas
+- Turn that blank canvas into desired output
+
+Error checking is necessary throughout this process because of various potential visual issues every time a new object is placed onto the canvas. Some examples of errors that may occur are overlapping elements and common object-oriented errors like null pointers and out-of-bound indexes. Since this DSL will rely on the IDE, errors will be shown in the terminal. A logging interface within this DSL will also print more human-readable errors in the terminal.
+
+
 ### Expressiveness
 
 _What should be easy to do in this language? What should be possible, but
 difficult? What should be impossible or very difficult?_
 
+Things that should be easy to do in this language are compositing images with text and creating multiple images. Possible, but difficult things to do in this language is any sort of text manipulation. The DSL should assume that the set of texts the user uploads at first is mostly final but allow for minor changes like spelling or punctuation errors. One function that should definitely be impossible is writing recursive loops. However, because this idea relies on the host language, I am not exactly sure how I can manually stop recursive loops. Perhaps this is adding some sort of detection in the compiler or a runtime restriction?
+
 ### Related work
 
 _Are there any other DSLs in this domain? If not, describe how you know there
 aren't and conjecture why not. If so, describe them and provide links. How well
 do they address the need? Are there any particularly admirable qualities of the
 language? Are there parts of the language you think could be improved?_
 
+ContextFree is an example of an image-generating DSL. Even though it seems likely, I was not able to find a DSL specifically for image composition. Maybe I wasn’t using the right keywords, but one possible reason may be that people who generally create the things this DSL makes are familiar with design tools like Adobe Photoshop and/or are unfamiliar with coding in general.
```

Comment:
I also had trouble finding DSLs for my topics. Even though this isn't a DSL (at least I don't think it is), I think there are AI tools that solve problems in this problem domain